app/src/main/java/org/projectbuendia/client/ui/chart/ChartRenderer.java

```diff
@@ -116,62 +127,100 @@ public void render(Chart chart, Map<String, ObsValue> latestObservations,
             for (ChartSection tileGroup : chart.tileGroups) {
                 List<Tile> tileRow = new ArrayList<>();
                 for (ChartItem item : tileGroup.items) {
-                    ObsValue[] obsValues = new ObsValue[item.conceptUuids.length];
-                    for (int i = 0; i < obsValues.length; i++) {
-                        obsValues[i] = latestObservations.get(item.conceptUuids[i]);
+                    ObsPoint[] points = new ObsPoint[item.conceptUuids.length];
+                    for (int i = 0; i < points.length; i++) {
+                        Obs obs = latestObservations.get(item.conceptUuids[i]);
+                        if (obs != null) {
+                            points[i] = obs.getObsPoint();
+                        }
+                    }
+                    tileRow.add(new Tile(item, points));
+                    if (!item.script.trim().isEmpty()) {
+                        mConceptsToDump.addAll(Arrays.asList(item.conceptUuids));
```

Comment:
Oops - just read the line above; I see that is what it currently is doing. Great!